TodoApp/Src/TodoApp.Api/Controllers/ItemsController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Microsoft.Web.Http;
+using TodoApp.Api.Services;
+using TodoApp.Contracts.Models;
+using TodoApp.Contracts.Repositories;
+using TodoApp.Contracts.Services;
+
+namespace TodoApp.Api.Controllers
+{
+    [ApiVersion("1.0")]
+    [RoutePrefix("api/v{version:apiVersion}/items")]
+    public class ItemsController : ApiController
+    {
+        private readonly IItemRepository _repository;
+        private readonly IUrlService _urlService;
+
+        public ItemsController(IItemRepository repository, IUrlService urlService)
+        {
+            _repository = repository;
+            _urlService = urlService;
+        }
+
+        // GET: api/v1/items
+        [Route]
+        public async Task<IHttpActionResult> GetAllItemsAsync()
+            => Ok(await _repository.GetAllAsync());
+
+        // GET: api/v1/items/5
+        [Route("{id}", Name = UrlService.NewItemRouteName)]
+        public async Task<IHttpActionResult> GetItemByIdAsync(Guid id)
+            => Ok(await _repository.GetByIdAsync(id));
+
+        // POST: api/v1/items
+        [Route("")]
+        public async Task<IHttpActionResult> PostItemAsync([FromBody]Item item)
+        {
+            var createdItem = await _repository.CreateAsync(item);
+            var url = _urlService.GetItemUrl(createdItem.Id);
```

Comment:
It is not completely necessary, however, I would like to know about a habbit that might ease reading of (your) code a bit: "If you use constructor or any other similar construct clearly stating returned type, use `var`. Otherwise, use `type` instead of `var`" – I'm almost sure there will be a war somewhere online and I'm not taking a side here, just informing you the posibility…